### PR TITLE
Add more metadata. Also additional memory stats

### PR DESCRIPTION
**Pull request by kylebrandt**
bf1 = "Back Fill 1". Adding various metadata. Also adding additional memory stats

### CODE REVIEW

File: collectors/procstats_linux.go [hunk 1]
```diff
@@ -65,27 +65,32 @@ func c_procstats_linux() (opentsdb.MultiDataPoint, error) {
...
+	Add(&md, osMemFree, int(mem["MemFree"])*1024, nil, metadata.Gauge, metadata.Bytes, "")
+	Add(&md, osMemUsed, (int(mem["MemTotal"])-(int(mem["MemFree"])+int(mem["Buffers"])+int(mem["Cached"])))*1024, nil, metadata.Gauge, metadata.Bytes,
+		"Total Memory - Free Memory+Cached+Buffers")
 	if mem["MemTotal"] != 0 {
-		Add(&md, osMemPctFree, (mem["MemFree"]+mem["Buffers"]+mem["Cached"])/mem["MemTotal"]*100, nil, metadata.Unknown, metadata.None, "")
+		Add(&md, osMemPctFree, (mem["MemFree"]+mem["Buffers"]+mem["Cached"])/mem["MemTotal"]*100, nil, metadata.Gauge, metadata.Pct,
+			"Percent of Memory Free + Cached _ Buffers")
 	}
-
+	// vmstat_desc := map[string]string{
```
Comment (madelynnblue) [inline collectors/procstats_linux.go]: What's this here for?

File: collectors/procstats_linux.go [hunk 2]
```diff
@@ -43,8 +43,8 @@ func c_procstats_linux() (opentsdb.MultiDataPoint, error) {
 		if m == nil {
 			return nil
 		}
-		Add(&md, "linux.uptime_total", m[1], nil, metadata.Unknown, metadata.None, "")
-		Add(&md, "linux.uptime_now", m[2], nil, metadata.Unknown, metadata.None, "")
+		Add(&md, "linux.uptime_total", m[1], nil, metadata.Counter, metadata.Second, "total number of seconds the system has been up")
+		Add(&md, "linux.uptime_idle", m[2], nil, metadata.Counter, metadata.Second, "total number of seconds the systemhas spent idle,")
```
Comment (madelynnblue) [inline collectors/procstats_linux.go]: Typo "systemhas"

File: collectors/procstats_linux.go [hunk 3]
```diff
@@ -65,27 +65,32 @@ func c_procstats_linux() (opentsdb.MultiDataPoint, error) {
...
-	Add(&md, osMemFree, int(mem["MemFree"])*1024, nil, metadata.Unknown, metadata.None, "")
-	Add(&md, osMemUsed, (int(mem["MemTotal"])-(int(mem["MemFree"])+int(mem["Buffers"])+int(mem["Cached"])))*1024, nil, metadata.Unknown, metadata.None, "")
+	Add(&md, osMemTotal, int(mem["MemTotal"])*1024, nil, metadata.Gauge, metadata.Bytes, "")
+	Add(&md, osMemFree, int(mem["MemFree"])*1024, nil, metadata.Gauge, metadata.Bytes, "")
+	Add(&md, osMemUsed, (int(mem["MemTotal"])-(int(mem["MemFree"])+int(mem["Buffers"])+int(mem["Cached"])))*1024, nil, metadata.Gauge, metadata.Bytes,
+		"Total Memory - Free Memory+Cached+Buffers")
 	if mem["MemTotal"] != 0 {
-		Add(&md, osMemPctFree, (mem["MemFree"]+mem["Buffers"]+mem["Cached"])/mem["MemTotal"]*100, nil, metadata.Unknown, metadata.None, "")
+		Add(&md, osMemPctFree, (mem["MemFree"]+mem["Buffers"]+mem["Cached"])/mem["MemTotal"]*100, nil, metadata.Gauge, metadata.Pct,
+			"Percent of Memory Free + Cached _ Buffers")
```
Comment (madelynnblue) [inline collectors/procstats_linux.go]: This description is confusing. Also I think it's best to remove the description of the calculation that is performed and instead just have a description of the metric. Description should be a sentence ending with a period.

File: collectors/procstats_linux.go [hunk 4]
```diff
@@ -65,27 +65,32 @@ func c_procstats_linux() (opentsdb.MultiDataPoint, error) {
 	}); err != nil {
 		Error = err
 	}
-	Add(&md, osMemTotal, int(mem["MemTotal"])*1024, nil, metadata.Unknown, metadata.None, "")
-	Add(&md, osMemFree, int(mem["MemFree"])*1024, nil, metadata.Unknown, metadata.None, "")
-	Add(&md, osMemUsed, (int(mem["MemTotal"])-(int(mem["MemFree"])+int(mem["Buffers"])+int(mem["Cached"])))*1024, nil, metadata.Unknown, metadata.None, "")
+	Add(&md, osMemTotal, int(mem["MemTotal"])*1024, nil, metadata.Gauge, metadata.Bytes, "")
+	Add(&md, osMemFree, int(mem["MemFree"])*1024, nil, metadata.Gauge, metadata.Bytes, "")
+	Add(&md, osMemUsed, (int(mem["MemTotal"])-(int(mem["MemFree"])+int(mem["Buffers"])+int(mem["Cached"])))*1024, nil, metadata.Gauge, metadata.Bytes,
+		"Total Memory - Free Memory+Cached+Buffers")
```
Comment (madelynnblue) [inline collectors/procstats_linux.go]: Same as below: why are we including a description of the calculation? Having "-" and "+", it's not clear that the "-" is a separator and not a minus. Should be a full sentence.

File: collectors/procstats_linux.go [hunk 5]
```diff
@@ -65,27 +65,32 @@ func c_procstats_linux() (opentsdb.MultiDataPoint, error) {
...
 			mio := inoutRE.FindStringSubmatch(m[1])
 			if mio != nil {
-				Add(&md, "linux.mem."+mio[1], m[2], opentsdb.TagSet{"direction": mio[2]}, metadata.Unknown, metadata.None, "")
+				Add(&md, "linux.mem."+mio[1], m[2], opentsdb.TagSet{"direction": mio[2]}, metadata.Counter, metadata.None, "")
 			} else {
-				Add(&md, "linux.mem."+m[1], m[2], nil, metadata.Unknown, metadata.None, "")
+				Add(&md, "linux.mem."+m[1], m[2], nil, metadata.Counter, metadata.None, "")
 			}
+		default:
+			Add(&md, "linux.mem."+m[1], m[2], nil, metadata.Counter, metadata.None, "")
```
Comment (madelynnblue) [inline collectors/procstats_linux.go:90]: Are you sure all other metrics, and possible future metrics, are counters? May be safer to specify those in another case.

File: collectors/procstats_linux.go [hunk 6]
```diff
@@ -118,7 +135,9 @@ func c_procstats_linux() (opentsdb.MultiDataPoint, error) {
 				if tag_cpu != "" {
 					tags["cpu"] = tag_cpu
 				}
-				Add(&md, "linux.cpu"+metric_percpu, value, tags, metadata.Unknown, metadata.None, "")
+				var desc string
+				desc, _ = cpu_stat_desc[CPU_FIELDS[i]]
+				Add(&md, "linux.cpu"+metric_percpu, value, tags, metadata.Counter, metadata.CHz, desc)
```
Comment (madelynnblue) [inline collectors/procstats_linux.go]: `Add(&md, "linux.cpu"+metric_percpu, value, tags, metadata.Counter, metadata.CHz, cpu_stat_desc[CPU_FIELDS[i]])` is all you need

File: collectors/procstats_linux.go [hunk 7]
```diff
@@ -43,8 +43,8 @@ func c_procstats_linux() (opentsdb.MultiDataPoint, error) {
 		if m == nil {
 			return nil
 		}
-		Add(&md, "linux.uptime_total", m[1], nil, metadata.Unknown, metadata.None, "")
-		Add(&md, "linux.uptime_now", m[2], nil, metadata.Unknown, metadata.None, "")
+		Add(&md, "linux.uptime_total", m[1], nil, metadata.Counter, metadata.Second, "total number of seconds the system has been up")
```
Comment (madelynnblue) [inline collectors/procstats_linux.go]: Descriptions are sentences ended with periods.

File: collectors/procstats_linux.go [hunk 8]
```diff
@@ -205,7 +239,9 @@ func c_procstats_linux() (opentsdb.MultiDataPoint, error) {
 				return fmt.Errorf("interrupts: unexpected value: %v", val)
 				break
 			}
-			Add(&md, "linux.interrupts", val, opentsdb.TagSet{"type": irq_type, "cpu": strconv.Itoa(i)}, metadata.Unknown, metadata.None, "")
+			var desc string
+			desc, _ = irq_type_desc[irq_type]
+			Add(&md, "linux.interrupts", val, opentsdb.TagSet{"type": irq_type, "cpu": strconv.Itoa(i)}, metadata.Gauge, metadata.Interupt, desc)
```
Comment (madelynnblue) [inline collectors/procstats_linux.go]: Same shortening as above.

File: collectors/procstats_linux.go [hunk 9]
```diff
@@ -93,6 +98,18 @@ func c_procstats_linux() (opentsdb.MultiDataPoint, error) {
 	}
 	num_cores := 0
 	var t_util float64
+	cpu_stat_desc := map[string]string{
+		"user":       "normal processes executing in user mode",
```
Comment (madelynnblue) [inline collectors/procstats_linux.go]: Descriptions start with a capital, end with a period.

File: collectors/procstats_linux.go [hunk 10]
```diff
@@ -159,21 +179,35 @@ func c_procstats_linux() (opentsdb.MultiDataPoint, error) {
...
 	}
 	if err := readLine("/proc/sys/kernel/random/entropy_avail", func(s string) error {
-		Add(&md, "linux.entropy_avail", strings.TrimSpace(s), nil, metadata.Unknown, metadata.None, "")
+		Add(&md, "linux.entropy_avail", strings.TrimSpace(s), nil, metadata.Gauge, metadata.Entropy, "The remaing amount of entropy available to the system. If it is low or hitting zero processes might be blocked waiting for extropy")
 		return nil
 	}); err != nil {
 		Error = err
 	}
+	irq_type_desc := map[string]string{
+		"NMI": "Non-maskable interrupts",
```
Comment (madelynnblue) [inline collectors/procstats_linux.go]: Same as other _desc map.

File: collectors/procstats_linux.go (hunk 5 above)
Comment (kylebrandt) [inline collectors/procstats_linux.go:90]: Would be, but they all look like counters and there are over 90 entries, so I'm okay with that for now

**Pull request merged**